### ISSUE TITLE
Shouldn't the wireName key be used when visiting an array?

### ISSUE THREAD

**Issue by my2ter** [guzzle/guzzle-services#151]
https://github.com/guzzle/guzzle-services/blob/0a8b5d304305a36b7a907c8ccf4971f232c54832/src/ResponseLocation/JsonLocation.php#L105

This seems better to me:
```php
public function visit(
    ResultInterface $result,
    ResponseInterface $response,
    Parameter $param
) {
$name = $param->getName();
$key = $param->getWireName();

// Check if the result should be treated as a list
if ($param->getType() == 'array') {
    // Treat as javascript array
    if ($name) {
        // name provided, store it under a key in the array
        $subArray = isset($this->json[$key]) ? $this->json[$key] : null;
        $result[$name] = $this->recurse($param, $subArray);
....
```

That would fix my problem but maybe my model is not properly defined... Even so, if that's the later, I don't understand the use of the wireName then.

Cheers.

**Comment by my2ter** [comment]
Hi @Konafets , sorry to bother you, could you have a look please?

**Comment by Konafets** [comment]
Hi @my2ter,

I never used that feature by myself. Did you consult the manual entry for this? There is a description of `sentAs`at http://guzzle3.readthedocs.io/webservice-client/guzzle-service-descriptions.html#parameter-schema

**Comment by my2ter** [comment]
@konafets the doc seems to be about a request but I'm only seeing a problem from the response POV. Check my PR #152 maybe, I made a test to explain the issue. I tagged you because you were the most recent person involved in the project as it seems to be a bit dead atm :) Sorry again for bothering you. Ta

**Comment by my2ter** [comment]
@Konafets I can confirm the behaviour I am describing based on the doc you listed:

> Response Models
> 
> Response models describe how a response is parsed into a Guzzle\Service\Resource\Model object. Response models are always modeled as JSON schema objects. When an HTTP response is parsed using a response model, the rules specified on each property of a response model will translate 1:1 as keys in a PHP associative array. **When a sentAs attribute is found in response model parameters, the value retrieved from the HTTP response is retrieved using the sentAs parameter but stored in the response model using the name of the parameter**.

Without my fix this behaviour isn't working as expected.

**Comment by my2ter** [comment]
Thanks @Konafets, can close this issue. Merged with #152
